### PR TITLE
Let engineering goggles (renamed to mesons) see through walls

### PR DESCRIPTION
**Pull request by notquitehadouken**
## About the PR

mesons now let you see certain things through walls

namely anything that isn't
* mobs
* items
* syndicate bombs
* power sinks
* lockers

## Why / Balance

it's cool

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: notquitehadouken (freshlysqueezedpain)
- add: Engineering goggles now let you see certain things through walls.

### CODE REVIEW

**Comment by notquitehadouken** [comment]
what feature freeze i can finish this in 4 days HAHAHAHAHA

**Comment by MACMAN2003** [comment]
will you rename the goggles back to meson scanners?

**Comment by notquitehadouken** [comment]
> will you rename the goggles back to meson scanners?

should i?

**Comment by MACMAN2003** [comment]
> > will you rename the goggles back to meson scanners?
> 
> should i?

iirc mesons have been synonymous with "seeing walls through walls" so yeah

**Comment by FungiFellow** [comment]
Time to combine this with a T-Ray for Engi Omnisience

File: Resources/Textures/Shaders/brightnessfilter.swsl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+uniform highp float THRESHHOLD;
+uniform sampler2D SCREEN_TEXTURE;
+
+void fragment() {
+    highp vec4 THISCOLOR = texture(SCREEN_TEXTURE, UV);
+    COLOR = vec4(1, 1, 1, 1);
```
Comment (deathride58) [inline Resources/Textures/Shaders/brightnessfilter.swsl:6]: This isn't going to compile on drivers that strictly comply with the GLES spec (such as the engine's built-in compatibility mode, which uses ANGLE, a compatibility layer that has very strict compliance with the GLES spec). These need to be made explicit floats, as implicit conversions from integer to float simply do not exist in the GLES spec.

```suggestion
    COLOR = vec4(1.0, 1.0, 1.0, 1.0);
```

File: Resources/Textures/Shaders/brightnessfilter.swsl [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    if (dot(THISCOLOR.rgb, THISCOLOR.rgb) <= THRESHHOLD)
+        COLOR = vec4(0, 0, 0, 1);
```
Comment (deathride58) [inline Resources/Textures/Shaders/brightnessfilter.swsl:8]: Ditto

```suggestion
        COLOR = vec4(0.0, 0.0, 0.0, 1.0);
```

File: Resources/Textures/Shaders/scanline.swsl [hunk 3]
```diff
@@ -0,0 +1,5 @@
+uniform highp vec4 OVERLAY_COLOR;
+
+void fragment() {
+    COLOR = vec4(vec3(mod(FRAGCOORD.y + floor(TIME * 30), 2)), 1) * OVERLAY_COLOR;
```
Comment (deathride58) [inline Resources/Textures/Shaders/scanline.swsl:4]: Ditto
```suggestion
    COLOR = vec4(vec3(mod(FRAGCOORD.y + floor(TIME * 30.0), 2.0)), 1.0) * OVERLAY_COLOR;
```

**Review by deathride58 — CHANGES_REQUESTED** [review]
The original post additionally needs footage of the new mesons in action.

File: Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml [hunk 4]
```diff
@@ -55,14 +55,16 @@
...
-  name: engineering goggles #less confusion
-  description: Green-tinted goggles using a proprietary polymer that provides protection from eye damage of all types.
+  name: mesons #less confusion
+  description: Green-tinted goggles that use a proprietary material that provides protection from eye damage and allows you to see through walls.
   components:
   - type: Sprite
     sprite: Clothing/Eyes/Glasses/meson.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Glasses/meson.rsi
   - type: EyeProtection
```
Comment (Lamrr) [inline Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:65]: Perhaps outside the scope of this PR, but now that these are mesons they should stop being welding goggles so engineers actually use welding helmets (or goggles if later added)

**Comment by notquitehadouken** [comment]
> The original post additionally needs footage of the new mesons in action.

i will do this once they function

File: Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml (hunk 4 above)
Comment (notquitehadouken) [reply Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:76]: this is a balance issue and someone can fix it in the ~~~future~~~ if it ends up being a problem worth addressing

**Comment by deltanedas** [comment]
that should have an accessibility toggle and/or only be used for out of view things, its really jarring

**Comment by FairlySadPanda** [comment]
image

**Comment by notquitehadouken** [comment]
image
i like this better anyway

**Comment by FungiFellow** [comment]
>I like this better anyway

Assuming that this is a toggle I like the gameboy green tint purely because it helps with low light and discourages having it on 24/7

**Comment by notquitehadouken** [comment]
> Assuming that this is a toggle, I like the gameboy green tint purely because it helps with low light and discourages having it on 24/7

lighting is disabled :godmode:

**Comment by Hmeister-real** [comment]
> image i like this better anyway

having no lighting is literally so based, low end people can now run an easy 60 fps with the gameboy vision

**Comment by FungiFellow** [comment]
One issue I immediately spot is that I can't see any wires in those maints

**Comment by notquitehadouken** [comment]
image


> One issue I immediately spot is that I can't see any wires in those maints

that is because there are no wires in those maints

**Comment by notquitehadouken** [comment]
i did not complete it in time :trollface:

**Comment by notquitehadouken** [comment]
i'll reopen it once the feature freeze is over

**Comment by notquitehadouken** [comment]
question: would this even get merged if i did complete it

**Comment by mirrorcult** [comment]
in the state last posted i imagine no, its just free wallhacks

**Comment by WarMechanic** [comment]
i think it will be fine if its balanced such that you can only see structural things or anything visible with a t-ray

**Comment by Dutch-VanDerLinde** [comment]
@notquitehadouken yo can I use this code for RMC

**Comment by notquitehadouken** [comment]
> @notquitehadouken yo can I use this code for RMC

dew it

**Pull request closed without merging**